### DISCUSSION TITLE
The quality of Text Editing and Rendering in Flutter

### DISCUSSION THREAD

**Discussion by osaxma** [flutter/uxr#81] [Research Ideas]
Honestly, I'm not sure where to start but I believe any Flutter app can be easily identified through a TextField as shown in this image:

<img width="1013" alt="Screen Shot 2021-12-28 at 3 40 47 AM" src="https://user-images.githubusercontent.com/46427323/147516469-837d0fce-0357-4d65-885f-34d1c1b1c662.png">

The image above has at least three obvious issues:
- Inconsistent Text Highlight [flutter#68675](https://github.com/flutter/flutter/issues/68675) && [flutter#39755](https://github.com/flutter/flutter/issues/39755)
- Lack of native selection controls [flutter#82907](https://github.com/flutter/flutter/issues/82907)
- Misplacement of selection handles _(I'm sure there's an issue but I couldn't find it)_

Other prominent issues:
- Selection behavior doesn't match native behavior (e.g. triple click to select all) [flutter#92966](https://github.com/flutter/flutter/issues/92966)
- Auto detect Text Directionality based on input language [flutter#91738](https://github.com/flutter/flutter/issues/91738)
- Ability to have multiple Text Directions within a text field [flutter#79749](https://github.com/flutter/flutter/issues/79749) such as:
<p align="center"><img width="500" alt="Screen Shot 2021-12-28 at 3 40 47 AM" src="https://user-images.githubusercontent.com/46427323/147517505-47806882-4df0-468c-99c4-fb2689a32522.png"></p>

- https://github.com/flutter/flutter/issues/28894 
- https://github.com/flutter/flutter/issues/12675

And many other issues under the following tags:
- [a: typography](https://github.com/flutter/flutter/labels/a%3A%20typography)
- [a: text input ](https://github.com/flutter/flutter/labels/a%3A%20text%20input)
- [a: internationalization](https://github.com/flutter/flutter/labels/a%3A%20internationalization)
- [a: accessibility](https://github.com/flutter/flutter/labels/a%3A%20accessibility)

---- 

I hope this area will be considered as a research topic with the simple goal of: 

- **Matching the native behavior and capabilities as much as possible.**

**Comment by jayoung-lee** [comment]
Hi @osaxma, thanks for starting this discussion. 

We are aware of the fidelity issues wrt text editing, and actually, we recently ran a user survey (in Q4 2021) asking feedback around this very topic! 

Based on the feedback we collected from the survey, the Core Framework team is highly focused on acting on it in 2022, especially the right-to-left/bi-directional text.

Now we know better about which issues need to be tackle, I'd entrust the task to the engineering team. @justinmc will be able to answer further questions, as he's working on a lot of these things. 

Thanks!

**Reply by jayoung-lee** [reply]
In case you're wondering, there is a Medium article about the Q4 2021 user survey published, covering this topic (text editing) along with others: https://medium.com/@jayoung.lee/q4-2021-user-survey-results-about-null-safety-ecosystem-text-editing-and-mobile-development-5d33341954e9

**Reply by osaxma** [reply]
@jayoung-lee thank you 🙏🏽

**Reply by alwijein** [reply]
thank you

**Comment by maximveksler** [comment]
Hey guys, a progress update would be truly helpful

**Comment by HansMuller** [comment]
CC @justinmc

**Comment by justinmc** [comment]
I have a PR in review that will refactor the context menus: https://github.com/flutter/flutter/pull/107193.  It won't fix most of the context menu problems in this post but it will make them easier to fix.

@Renzo-Olivares is working on a project (https://github.com/flutter/flutter/pull/105165) that will fix the selection behavior issues mentioned like https://github.com/flutter/flutter/issues/92966.

We're doing a better job of tracking the RTL problems thanks to the work that @jayoung-lee mentioned above.  I'm hoping we can start fixing some of those.

Overall we have a lot of work here, but almost everything in this post has a viable solution.  Anyone that wants to help contribute can feel free to jump on any of the linked issues that don't have an assignee.

**Reply by victorhtrabajo** [reply]
Thanks.

**Comment by alwijein** [comment]
can i copy just one click button or any
